src/rebuild/scan.c

```diff
@@ -647,10 +716,11 @@ rebuild_obj_scan_cb(daos_handle_t ch, vos_iter_entry_t *ent,
 	if (map != NULL)
 		pl_map_decref(map);
 
-	if (--arg->yield_freq == 0) {
+	if (--arg->yield_freq == 0 || --arg->obj_yield_cnt == 0) {
```

Comment:
why need obj_yield_cnt besides yield_freq? and seems obj_yield_cnt possible become < 0 here as above for rebuild_nr do obj_yield_cnt--?